doc/wg/core/notes/core-notes-2024-10-11.md

```diff
@@ -0,0 +1,88 @@
+ * Brad: So, now I've been looking into machinery for components to standardize configuring boards. Still iterating on this with new ideas.
+ * Leon: I have been trying to make it so the core kernel logic doesn't need to care about process-specific debugging info. That created so many issues requiring generic types literally everywhere in the codebase. It's still worth pursuing disentangling process debugging and the core kernel, but this PR is still a great first step.
+ * Brad: What I've found is that types are tractable as long as everything is for ProcessStandard, but not for other things. That does make different implementations more complicated, but simplifies most of our code
+### Nightly Update
+ * https://github.com/tock/tock/pull/4193
+ * Brad: This PR has some static mut issues. Amit says they're "easy to handle" and he'll help.
+ * Hudson: We could also make a partial jump if we needed, to somewhere less than here
+ * Leon: The warnings in this nightly are good, but they haven't reached the level where there are a TON of them. And they occur in tests
+ * Hudson: Notably, I think warnings in tests don't block the build right now
+ * Leon: Oh, that's on me to get to
+
+
+## Treadmill Updates
+ * https://github.com/treadmill-tb/tock/pull/2
+ * Ben: I have a PR in progress that makes a testing framework for adapting existing Tock tests into Treadmill. It defines a board file and a testing framework file in Python. The Board file tells you how to get the UART and Board for programming. The framework shows the tests to run
+ * Ben: The board harness has architecture, path in the Tock repo, ability to get UART port, and functions for flashing it
+ * Ben: The testing framework such as the "c_hello.py" file, has a test, steps to load it on the board such as flashing the kernel and app or apps. Then for oneshot tests, which are the typical ones, they run once and collect info
+ * Ben: Two types of tests, AnalyzeConsole expects output from serial ports. There's also a WaitForConsoleMessage test which instead just waits for a single message.
+ * Ben: This can run on real hardware with Treadmill now. In the test-execute for Treadmill, the runner will load things, wait for a message, and then return success/failure
+ * Leon: The basic idea here is that for any given test we want to combine a specification for interaction with a board with a specification for how the test works. So the board is interaction and the test is the strategy of when to flash things, when to flush buffers, when to expect output, when to use GPIO pins, etc. We can inherit parts of this to abstract away common choices here that many tests use. For example WaitForConsoleMessage is an abstraction that just takes an app and a string.
+ * Leon: So the goal here was to balance expressibility and abstracting detail away for simple tasks
```

Comment:
```suggestion
 * Leon: So the goal here was to balance expressiveness and abstracting detail away for simple tasks
```
Probably misspoke there. :smile:

Reply:
In my head the two words mean exactly the same thing. So I'd blame me for this one.